examples/defaulter-gen/generators/defaulter.go

```diff
@@ -432,19 +450,30 @@ func buildCallTreeForType(t *types.Type, root bool, existingDefaulters, newDefau
 	// base has been added already, now add any additional defaulters defined for this object
 	parent.call = append(parent.call, defaults.additional...)
 
+	// if the type already exists, don't build the tree for it and don't generate anything.
+	// This is used to avoid recursion for nested recursive types.
+	if c.currentlyBuildingTypes[t] {
+		return nil
+	}
+	// if type doesn't exist, mark it as existing
+	c.currentlyBuildingTypes[t] = true
+
 	switch t.Kind {
 	case types.Pointer:
-		if child := buildCallTreeForType(t.Elem, false, existingDefaulters, newDefaulters); child != nil {
+		r := newCallTreeForType(c.existingDefaulters, c.newDefaulters, c.currentlyBuildingTypes)
```

Comment:
What I was trying to communicate is that you *don't* call newCallTreeForType here, you reuse `c`.